File: src/main/java/com/coffee/shop/api/demo/product/domain/api/dto/ProductDto.java
package com.coffee.shop.api.demo.product.domain.api.dto;

import com.coffee.shop.api.demo.image.api.dto.ImageDto;
import java.math.BigDecimal;
import java.util.List;
import java.util.Map;
import lombok.Data;

@Data
public class ProductDto {

    private Long id;
    private String name;
    private String description;
    private String colorCode;
    private BigDecimal price;
    private String categoryName;
    private String categoryCode;
    private Map<String, String> propertiesValue;
    private List<ImageDto> listImages;
    private List<ImageDto> cardImages;
}

File: src/main/kotlin/com/coffee/shop/api/demo/product/domain/api/dto/ProductDto.kt
package com.coffee.shop.api.demo.product.domain.api.dto

import com.coffee.shop.api.demo.image.api.dto.ImageDto
import lombok.Data
import java.math.BigDecimal

@Data
class ProductDto {
    var id: Long? = null
    var name: String? = null
    var description: String? = null
    var colorCode: String? = null
    var price: BigDecimal? = null
    var categoryName: String? = null
    var categoryCode: String? = null
    var propertiesValue: Map<String, String>? = null
    var listImage: ImageDto? = null
    var cardImages: List<ImageDto>? = null
}